### ISSUE TITLE
ConvertString should return Failure(...) when a string ctor fails instead of propagating the exception.

### ISSUE THREAD

**Issue by kenmcgowan** [dotnet/command-line-api#696]
Currently, when an argument is bound to a type with a constructor that accepts a string, if that constructor throws an exception (for example, if validation fails), the exception is propagated and a stack trace is displayed on the command line.

Compare this to when conversion fails using the same constructor, but from within a custom `TypeConverter`. Instead of propagating the exception, the `ConvertString` method returns `Failure(...)` and a much more user-friendly message is displayed on the command line.

This section has no exception handling:
https://github.com/dotnet/command-line-api/blob/d9fea6f36bf23fa174656bb5e8299c03c07b5494/src/System.CommandLine/ArgumentConverter.cs#L77-L82

vs. this, in the same method, which handles exceptions thrown during type conversion:
https://github.com/dotnet/command-line-api/blob/d9fea6f36bf23fa174656bb5e8299c03c07b5494/src/System.CommandLine/ArgumentConverter.cs#L61-L70

It would be preferable if both scenarios handled exceptions and generated more user-friendly responses.

**Comment by jonsequitur** [comment]
I agree that this would be an improvement. The approach we've taken is that the developer should ideally handle this at the validation stage because it will be hard for the end user to understand how to fix the input. So the audience for this error output is currently the developer, not the end user. 

Any suggestions for how to generalize this?

**Comment by kenmcgowan** [comment]
That makes sense. Being able to provide better guidance via validation is best in the long run.

For the case where validation isn't implemented yet, I think consistency would provide a better default experience. E.g., wrap the constructor invocation in a try/catch and return a `Failure(...)` when an exception is thrown. This better allows me to implement my CLI incrementally. I.e., I can defer customizing messages until the CLI design stabilizes.

The default messages generated on failure may not be detailed, but they at least tell the user "I can't parse that value," whereas the stack trace seems to say, "I'm broken, this is a defect".
